FWCore/Framework/src/EventProcessor.cc

```diff
@@ -397,7 +388,24 @@ namespace edm {
     unsigned int nConcurrentLumis =
         optionsPset.getUntrackedParameter<unsigned int>("numberOfConcurrentLuminosityBlocks");
     if (nConcurrentLumis == 0) {
-      nConcurrentLumis = nConcurrentRuns;
+      nConcurrentLumis = 2;
+    }
+    if (nConcurrentLumis > nStreams) {
+      nConcurrentLumis = nStreams;
+    }
+    std::vector<std::string> loopers = parameterSet->getParameter<std::vector<std::string>>("@all_loopers");
+    if (!loopers.empty()) {
+      //For now loopers make us run only 1 transition at a time
+      nStreams = 1;
+      nConcurrentLumis = 1;
+      nConcurrentRuns = 1;
+    }
```

Comment:
We should probably add a `LogWarning` if this changes the settings. I know we didn't have that before, but that was probably bad.

Reply:
Done, I added the LogWarning. Thanks.